doc/manual/src/contributing/hacking.md

```diff
@@ -132,6 +111,26 @@ $ nix-build
 
 You can also build Nix for one of the [supported platforms](#platforms).
 
+## Makefile variables
+
+You may need `profiledir=$out/etc/profile.d` and `sysconfdir=$out/etc` to run `make install`.
+
+Run `make` with [`--environment-overrides`](https://www.gnu.org/software/make/manual/make.html#index-_002de) to allow environment variables to override `Makefile` variables:
```

Comment:
They don't recommend this. I think we're supposed to pass them on the command line. If your command line has a bunch of `make foo && make bar` though, I think this is indeed worth the "risk", but considering that this is just for convenience, we should suggest `-e`. That long name is way too long to use more than once in a combined shell line.

Reply:
Many things have `?=` definitions which already get overridden, too.

Reply:
(we replied concurrently) I think @roberth meant don't do `-e` *or* `--environment-overrides`.